Vagrantfile

```diff
@@ -0,0 +1,22 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+# All Vagrant configuration is done below. The "2" in Vagrant.configure
+# configures the configuration version (we support older styles for
+# backwards compatibility). Please don't change it unless you know what
+# you're doing.
+Vagrant.configure(2) do |config|
+  config.vm.box = "ubuntu/trusty64"
+  config.vm.network "forwarded_port", guest: 8123, host: 8123
+  config.vm.synced_folder ".", "/vagrant"
+  config.vm.provision "shell", inline: <<-SHELL
```

Comment:
To my eyes, too many advertised options for setting up a development environment could be confusing for the newcomer and, most importantly, hard to maintain properly.
Personally I'd like to see only one _suggested_ (or _supported_?) setup that is as much portable as possible, and Vagrant seems to be a good fit for this given that's available for all the main desktop platforms.
In the end, developers confident enough in setting up their environments will always be free (and tempted) to chose whatever they prefer, and for them this version of Vagrantfile won't be very useful.
Those either new with development in general or new with HA will start most likely following what's written in the [Developers section](https://home-assistant.io/developers/development_environment/) of the website, and I think we should focus on making whatever's suggested there easy to setup and use as a fully capable testing environment (that to me means: easy to make changes + test them + run tests/linter/etc + create a commit to be sent for code review) instead of trying to provide every possible way out there that can be used to setup a dev environment.
Only my 2 cents, of course, I don't even have write access here so whatever I say is just for the sake of the argument, which is very interesting to me!